src/sandbox/index.js

```diff
@@ -1,13 +1,16 @@
-let db = require('./db')
-let env = require('./env')
-let events = require('./events')
-let http = require('./http')
-let start = require('./start')
+let sandbox = require('@architect/sandbox')
+let version = require('../../package.json').version
 
-module.exports = {
-  db,
-  env,
-  events,
-  http,
-  start,
+module.exports = function sandboxImplementation(opts) {
+  process.env.ARC_VERSION = `Architect ${version}`
+  let findPort = opt=> opt === '-p' || opt === '--port' || opt === 'port'
+  if (opts && opts.some(findPort)) {
+    if (opts.indexOf('-p') >= 0)
+      process.env.PORT = opts[opts.indexOf('-p') + 1]
+    if (opts.indexOf('--port') >= 0)
+      process.env.PORT = opts[opts.indexOf('--port') + 1]
+    if (opts.indexOf('port') >= 0)
+      process.env.PORT = opts[opts.indexOf('port') + 1]
```

Comment:
In terms of how the modules should integrate with the kitchen sink / parent module, I'd recommend not using env vars, and instead leverage function arguments. Consider the case where someone not at Begin ends up using the architect sub-modules for their own purposes - perhaps for their own CLI - and in that situation we would be forcing on them the technical decision to use env vars as the interface to the sub-module. If instead we leverage function arguments, then that gives consumers of the module freedom in how they want to interface with the sub-module (they could choose to translate env vars into function arguments, or use CLI arguments directly, or some other form of message passing).

This moment, when we are splitting up the modules, is a good time to decide on that approach and think about future compatibility as much as we can.

Reply:
Makes sense. `process.env.PORT` is legacy and we should retain support for it, but I agree that it'd be a good idea to start passing args – def true that now's the time to do that. Will factor that into new sandbox!